react-common/components/profile/UserAvatarDropdown.tsx

```diff
@@ -0,0 +1,53 @@
+import * as React from "react";
+import { MenuDropdown, MenuItem } from "../controls/MenuDropdown";
+import { classList } from "../util";
+
+interface IProps {
+    userProfile: pxt.auth.UserProfile;
+    title: string;
+
+    onSignOutClick?: () => void;
+    className?: string;
+    items?: MenuItem[];
+}
+
+export const UserAvatarDropdown: React.FC<IProps> = (props) => {
+    const { userProfile, title, items, onSignOutClick, className } = props;
+
+    const avatarUrl = userProfile?.idp?.pictureUrl ?? userProfile?.idp?.picture?.dataUrl;
```

Comment:
This will cause problems for Microsoft accounts. I updated this last week, but I just realized I didn't test this for clever accounts, so it's possible we need all of the methods. https://github.com/microsoft/pxt/pull/9956/files